### PR TITLE
Add QR-DQN support

### PR DESCRIPTION
**Pull request by ku2482**
Add QR-DQN support, related to this [issue](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib/issues/12) and [PR](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib/pull/13).

## Description
- [x] Add QR-DQN support.
- [x] Confirm the performances in `Pong` and `Breakout` are comparable with [coach](https://github.com/IntelLabs/coach/tree/master/benchmarks/qr_dqn).

Missing:
- [x] New release of SB3 contrib (so the CI passes)

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist:

- [ ] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [ ] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [ ] I have reformatted the code using `make format` (**required**)
- [ ] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [ ] I have ensured `make pytest` and `make type` both pass. (**required**)

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

**Comment by ku2482** [comment]
@araffin 
I have the computational resources available now, so I'm running the training in `Pong` and `Breakout`.
I'll let you know the results ;)

**Comment by araffin** [comment]
> I have the computational resources available now, so I'm running the training in Pong and Breakout.

I already started too ;) (but not with the same `exploration_fraction`, btw, why is it not the default?)

**Comment by ku2482** [comment]
Thanks :)

>but not with the same exploration_fraction, btw, why is it not the default?

`total_timesteps` is 5e7 for Atari experiment in the paper, so `exploration_fraction` is 0.005 and explore `5e7 * 0.005 = 2.5e5` steps by default. This example trains only 1e7 timesteps, so I set `exploration_fraction` to 0.025 (`1e7 * 0.025 = 2.5e5`).

**Comment by araffin** [comment]
>total_timesteps is 5e7 for Atari experiment in the paper, so exploration_fraction is 0.005 and explore 5e7 * 0.005 = 2.5e5 steps by default. This example trains only 1e7 timesteps, so I set exploration_fraction to 0.025 (1e7 * 0.025 = 2.5e5).

alright. We should probably make it the default to be consistent with SB3 DQN and how we benchmarked it.

**Comment by Miffyli** [comment]
Pitching in here, but I think for reliable comparison results we should, at least once, train on the very same settings as in the paper/reference implementation. Of course this could be the final training run after other runs seem to work out.

**Comment by ku2482** [comment]
@Miffyli 
I mentioned the differences between the evaluation protocols of sb3 and the paper below.
https://github.com/Stable-Baselines-Team/stable-baselines3-contrib/pull/13#issuecomment-742913092

If needed, I'd like to implement an eval callback for atari experiments as in the paper.
(Though, I have no computer resource for that long (about 1 week) evaluation.)

**Comment by Miffyli** [comment]
Ah okey, thanks for the pointer :). Good discussions there!

> If needed, I'd like to implement an eval callback for atari experiments as in the paper.
> (Though, I have no computer resource for that long (about 1 week) evaluation.)

I might be able to pitch in computing resources for this once code is ready (depends on the timing). I'd really like to see as-close-as-possible comparison with original hyperparameters in one or two games, as this integrity is main driving point of the contrib repo ^^.

**Comment by araffin** [comment]
>I might be able to pitch in computing resources for this once code is ready (depends on the timing). I'd really like to see as-close-as-possible comparison with original hyperparameters in one or two games, as this integrity is main driving point of the contrib repo ^^.

Also I agree with your statement, I would say I'm fairly confident we can reproduce the results, especially because QR-DQN and TQC share the most essential part, the huber loss ;) (and because TQC is working very well in my experiments, so even if there is a bug, it works ^^")

I also have results similar to intel coach (even with slightly different hyperparameters) for now.

**Review by araffin — APPROVED** [review]
